Tests/SentryTests/Integrations/ANR/SentryANRTrackingIntegrationTests.swift

```diff
@@ -405,7 +405,61 @@ class SentryANRTrackingIntegrationTests: SentrySDKIntegrationTestsBase {
             XCTAssertEqual(scope?.breadcrumbs().count, 0)
         }
     }
-    
+
+    func testV2Detected_PauseCalled_ANRStopped_DoesCaptureEvent() throws {
+        // Arrange
+        givenInitializedTracker(enableV2: true)
+        setUpThreadInspector()
+        Dynamic(sut).anrDetectedWithType(SentryANRType.fullyBlocking)
+
+        sut.pauseAppHangTracking()
+
+        // Act
+        let result = SentryANRStoppedResult(minDuration: 1.851, maxDuration: 2.249)
+        Dynamic(sut).anrStoppedWithResult(result)
+
+        // Assert
+        try assertEventWithScopeCaptured { event, scope, _ in
+            let ex = try XCTUnwrap(event?.exceptions?.first)
+            XCTAssertEqual(ex.mechanism?.type, "AppHang")
+            XCTAssertEqual(ex.type, "App Hang Fully Blocked")
+            XCTAssertEqual(ex.value, "App hanging between 1.9 and 2.2 seconds.")
+
+            // We use the mechanism data to temporarily store the duration.
+            // This asserts that we remove the mechanism data before sending the event.
+            let mechanismData = try XCTUnwrap(ex.mechanism?.data)
+            XCTAssertTrue(mechanismData.isEmpty)
+
+            XCTAssertNotNil(ex.stacktrace)
+            XCTAssertEqual(ex.stacktrace?.frames.first?.function, "main")
+            XCTAssertEqual(ex.stacktrace?.snapshot?.boolValue, true)
+            XCTAssertEqual(try XCTUnwrap(event?.threads?.first).current?.boolValue, true)
+            XCTAssertEqual(event?.isAppHangEvent, true)
+
+            let threads = try XCTUnwrap(event?.threads)
+
+            // Sometimes during tests its possible to have one thread without frames
+            // We just need to make sure we retrieve frame information for at least one other thread than the main thread
+            let threadsWithFrames = threads.filter {
+                ($0.stacktrace?.frames.count ?? 0) >= 1
+            }.count
```

Comment:
`m`: In case you know please add a comment _why_ there are threads without frames

Reply:
I'm sorry, I can't recall why that is. I copied the test and modified it a bit.